src/plugins/upload.js

```diff
@@ -0,0 +1,52 @@
+import tus from 'tus-js-client'
+
+export default {
+  install(Vue) {
+    Vue.mixin({
+      data() {
+        return {
+          chunkedUploadFilesInProgress: []
+        }
+      },
+
+      methods: {
+        uploadChunkedFile(file) {
+          const upload = new tus.Upload(file, {
+            endpoint: this.$client.helpers._webdavUrl,
+            headers: this.$client.helpers.buildHeaders(),
+            retryDelays: [0, 3000, 5000, 10000, 20000],
+            metadata: {
+              name: file.name,
+              type: file.type,
+              size: file.size
+            },
+
+            onError: error => {
+              console.error(error)
```

Comment:
reject the promise here ?

Reply:
I was also thinking about this but I want to test how does it behave when retrying the upload. AFAICT this will be triggered every time when the upload fails and rejecting it before retry feels a bit odd to me.